Assets/MixedRealityToolkit.Services/InputAnimation/Editor/InputRecordingWindow.cs

```diff
@@ -0,0 +1,351 @@
+            {
+                EditorGUILayout.HelpBox("Input test recording is only available in play mode", MessageType.Info);
+                return;
+            }
+            if (RecordingService == null)
+            {
+                EditorGUILayout.HelpBox("No input recording service found", MessageType.Info);
+                return;
+            }
+
+            using (new GUILayout.HorizontalScope())
+            {
+                bool newUseTimeLimit = GUILayout.Toggle(RecordingService.UseBufferTimeLimit, "Use buffer time limit");
+                if (newUseTimeLimit != RecordingService.UseBufferTimeLimit)
+                {
+                    RecordingService.UseBufferTimeLimit = newUseTimeLimit;
+                }
+
+                using (new GUIEnabledWrapper(RecordingService.UseBufferTimeLimit))
+                {
+                    float newTimeLimit = EditorGUILayout.FloatField(RecordingService.RecordingBufferTimeLimit);
+                    if (newTimeLimit != RecordingService.RecordingBufferTimeLimit)
+                    {
+                        RecordingService.RecordingBufferTimeLimit = newTimeLimit;
+                    }
+                }
+            }
+
+            bool wasRecording = RecordingService.IsRecording;
+            var recordButtonContent = wasRecording
+                ? new GUIContent(iconRecordActive, "Stop recording input animation")
+                : new GUIContent(iconRecord, "Record new input animation");
+            bool record = GUILayout.Toggle(wasRecording, recordButtonContent, "Button");
+
+            if (record != wasRecording)
+            {
+                if (record)
+                {
+                    RecordingService.StartRecording();
+                }
+                else
+                {
+                    RecordingService.StopRecording();
+
+                    SaveAnimation(true);
+                }
+            }
+
+            DrawAnimationInfo();
+        }
+
+        private void DrawPlaybackGUI()
+        {
+            if (!Application.isPlaying)
+            {
+                EditorGUILayout.HelpBox("Input test playback is only available in play mode", MessageType.Info);
+                return;
+            }
+            if (SimService == null)
+            {
+                EditorGUILayout.HelpBox("No input simulation service found", MessageType.Info);
+                return;
+            }
+
+            DrawAnimationInfo();
+
+            using (new GUILayout.HorizontalScope())
+            {
+                if (GUILayout.Button("Load ..."))
+                {
+                    string filepath = EditorUtility.OpenFilePanel(
+                        "Select input animation file",
+                        "",
+                        InputAnimationSerializationUtils.Extension);
+
+                    LoadAnimation(filepath);
+                }
+            }
+
+            using (new GUIEnabledWrapper(PlaybackService != null))
```

Comment:
@killerantz  FYI -- this is a pretty cool / clean way to show GUI as disabled if condition isn't satisfied (in this case PlaybackService != null) -- might be a nice way to clean up interactable. I didn't know about it until reading this PR, figured I would share.